relay-server/src/actors/processor.rs

```diff
@@ -2353,6 +2358,57 @@ mod tests {
             .unwrap()
     }
 
+    #[test]
+    fn test_it_keeps_or_drops_transactions() {
+        relay_test::setup();
+
+        // an empty json still produces a valid config
+        let json_config = serde_json::json!({});
+
+        let config = Config::from_json_value(json_config.clone()).unwrap();
+        let arconfig = Arc::new(Config::from_json_value(json_config).unwrap());
+
+        // it really shouldn't be necessary to start an entire service for a unit test
+        // it was ported from a python integration test
+        ServiceState::start(arconfig).unwrap();
```

Comment:
This starts all the actors/services we have in order for this unit test to work, which is kind of heavy and does not make much sense. 

But at this point I'm not sure how to solve this yet. 

The current status:
*  the function you test uses `reject` function on `envelope_context` which triggers the `TestStore` capture functionality 
https://github.com/getsentry/relay/blob/3b42784ba9359db65bd890b8d628414e3201de8c/relay-server/src/utils/envelope_context.rs#L190-L191

* this function also is triggered on `Drop` of the `envelope_context`
* also the `reject` function triggers `track_outcome` which also requires `Outcome` actor to be available in the registry.